File: tests/typesafety/typescript/unit_newtype.ts
// expect: passes

declare const secondsBrand: unique symbol;

type Seconds = number & { readonly [secondsBrand]: true };

function seconds(n: number): Seconds {
  return n as Seconds;
}

function waitBeforeRetry(delay: Seconds): string {
  return `sleeping ${delay}s`;
}

export function plan(): string {
  const timeout = seconds(30);
  return waitBeforeRetry(timeout);
}

File: tests/typesafety/python/unit_newtype.py
# expect: passes
# title: Seconds and Meters become types the checker can tell apart
# improves: alias_transparent
from typing import NewType

Seconds = NewType("Seconds", int)
Meters = NewType("Meters", int)


def wait_before_retry(delay: Seconds) -> str:
    return f"sleeping {delay}s"


def plan() -> str:
    timeout = Seconds(30)
    return wait_before_retry(timeout)
